### ISSUE TITLE
Gitlab changelog generation fails with 404 on 1.14.1

### ISSUE THREAD

**Issue by m4x1202** [goreleaser/goreleaser#3706]
### What happened?

I experience the same issue that is described here: https://github.com/goreleaser/goreleaser/issues/3358

The suggested solution there was to update to the latest version, which I am using (ref. `goreleaser --version` output below).

I checked the linked code pieces in https://github.com/xanzy/go-gitlab and checked the used version in [`go.mod`](https://github.com/goreleaser/goreleaser/blob/main/go.mod#L46), but everything seems to use up-to-date versions and from all that I can figure out the `PathEscape(project)` should be running.

The error log line that I am getting however suggests otherwise:
```
• generating changelog
⨯ release failed after 2m49s error=GET https://<privately_hosted_gitlab_url>/api/v4/projects/<project_path_not_urlencoded>/repository/compare: 404 {message: 404 Project Not Found}
```
`project_path_not_urlencoded = e.g. backend/test`

gitlab version: `15.6.2-ee`

### How can we reproduce this?

I used the config snippet from https://github.com/goreleaser/goreleaser/issues/3358#issue-1365695462

There is somewhat of a workaround for `release:`, by setting `gitlab.name` to the project_id, but this does not work for changelog unfortunately.
The only option I currently see is switching back to the default: `use: git`.

### goreleaser version

```bash
goreleaser version 1.14.1
commit: 7be70934534b5c9798b6ab1b7b3e5c35df7e8b2e
built at: 2023-01-02T13:53:59Z
built by: goreleaser
goos: darwin
goarch: amd64
module version: v1.14.1, checksum: h1:b+cW4BSDL5Ii32scSX8IoXrnpwHIItXgZyMskBhNK8s=

https://goreleaser.com
```

### GoReleaser Check

- [X] goreleaser check shows no errors

### Search

- [X] I did search for other open and closed issues before opening this.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by caarlos0** [comment]
maybe your gitlab token is invalid/don't have permissions?

**Comment by m4x1202** [comment]
Sorry about that, should have added this to my initial description.
I'm using `use_job_token: true`. As far as I know Gitlab's [`CI_JOB_TOKEN`](https://docs.gitlab.com/ee/ci/jobs/ci_job_token.html) should have enough permissions to pull the changelog, does it not?

**Comment by m4x1202** [comment]
To verify I additionally now tested with both a "Personal Access Token" and a "Project Access Token" with `api` Scope. Same error both times.

**Comment by m4x1202** [comment]
Did more debugging and of course your were absolutely right.

My last 2 comments were caused by fundamentally different "issues".

1. (my initial issue) To potentially prevent others from falling for the same trap: Turns out the gitlab `CI_JOB_TOKEN` actually has no permissions to make requests against the [Repository API](https://docs.gitlab.com/ee/api/repositories.html) (Side note: but it apparently has permissions to clone repos 😕; will check Gitlab if there already is an issue open for this).
2. (my incorrect verification of the issue) When using the "Personal"&"Project" tokens I ran `goreleaser` locally and with locally created tags (not pushed to gitlab!), thus of course Gitlab had no idea about them and correctly responded with 404. (I just thought it was the same issue, because of the same error code). Once I pushed them `goreleaser` and the `compare` command worked as expected.

Closing this issue. Thanks a lot for the support @caarlos0.
